arbnode/execution/sequencer.go

```diff
@@ -184,7 +184,7 @@ func (c *nonceCache) matches(header *types.Header) bool {
 	if c.dirty != nil {
 		// The header is updated as the block is built,
 		// so instead of checking its hash, we do a pointer comparison.
```

Comment:
We should update the comment here to note that even though the hash changes, c.dirty points to the header which changes so this check will pass nonetheless.

Reply:
Done.